### DISCUSSION TITLE
grub(?) error after update, can't boot Debian

### DISCUSSION THREAD

**Discussion by thebestcloudislocal** [SpiralLinux/SpiralLinux-project#303] [Q&A] [answered]
_(Crossposted to "Issues")_

Hello,

I recently "aptitude full-upgrade"d a Sid installation. I remember that a new kernel was installed and that snapshots were listed, I suppose this list was related to grub.

After booting the usual grub password ask dialog appears. It's possible to enter the password and slot opened appears, immediately followed by a reboot into the main board UEFI. I guess that's just the boot order.

Can you tell me how I could mount @home with an USB Linux, so that I could "rsync -av" my current /home folder (not a snapshot) to an external hard disk?

Also, is it easily possible to repair grub (I guess it's a grub error)?

Thanks in advance for any help you can provide!

**Comment by fweller** [comment, marked as answer]
This is an issue that happened to many users of the Sid repo recently.  There is some chatter about it on the Siduction forums.  I am using BtrFS with Snapper and had to boot from a Linux USB installer, then unlock my PC LUKS encryption, then mount the BtrFS filesystem, and then chroot into the PC, and then finally run the grub installer.  Below are my notes on the topic.

# Mount LUKS encrypted partition (if applicable)
cryptsetup luksOpen /dev/nvme0n1p2 crypt

# Create our chroot directory
mkdir /mnt/chroot

# Mount the installation partition (use only one of these options)
# 1 - No LUKS
mount /dev/sda2 /mnt/chroot
# 2 - LUKS
mount /dev/mapper/crypt /mnt/chroot
# 3 - BtrFS on top of LUKS
mount -o subvol=@ /dev/mapper/crypt /mnt/chroot
# 4 - BtrFS on top of LUKS & using zstd compression
mount -o compress=zstd,subvol=@ /dev/mapper/crypt /mnt/chroot

# Mount the boot partition (if it exists)
mount /dev/nvme0n1p1 /mnt/chroot/boot

# Mount the EFI partition
mount /dev/nvme0n1p1 /mnt/chroot/boot/efi

# Mount the target root filesystem
for i in /dev /dev/pts /proc /sys /sys/firmware/efi/efivars /run; do  mount -B $i /mnt/chroot$i; done
mv /mnt/chroot/etc/resolv.conf /mnt/chroot/etc/resolv.conf.bak 2>/dev/null
cp -a /etc/resolv.conf /mnt/chroot/etc/resolv.conf

# Chroot
chroot /mnt/chroot /bin/sh -c "mount -a || true && /bin/bash"

# Install Grub
grub-install

# Update Grub (probably unnecessary)
update-grub

**Reply by geckolinux** [reply]
Thanks very much for the excellent tips/instructions.

**Comment by fweller** [comment]
these two commands are probably unnecessary
mv /mnt/chroot/etc/resolv.conf /mnt/chroot/etc/resolv.conf.bak 2>/dev/null
cp -a /etc/resolv.conf /mnt/chroot/etc/resolv.conf

**Comment by thebestcloudislocal** [comment]
When I first tried it (with a Debian 12 Live USB stick) (I found the `/dev/` names with `lsblk`), there was an error message:

```
root@debian:~# chroot /mnt/chroot /bin/sh -c "mount -a || true && /bin/bash"
mount: /.snapshots: special device /dev/mapper/luks-82fd1d95-f7fb-447d-868c-72178f0dbc33 does not exist.
```
I rebooted and tried it again, but this time I started with:
`cryptsetup luksOpen /dev/sda2 luks-82fd1d95-f7fb-447d-868c-72178f0dbc33`

Thanks a lot. System is running again.

**Comment by Bengelsen** [comment]
Thank you! This saved my installation. Had to adjust a few of the commands to get i all back, and wanted to share.

I have a standard Spiral installation, BtrFS no LUKS, and running SID. Update a few days ago broke my boot. Throws me into the main board menu just like OP’s system. System set up with EFI partition. Spiral Linux installed on the nvme0n1 disk

I booted into a Debian 12.1 live KDE and ran these commands in Konsole:

user@debian:~$ `sudo su`
root@debian:/home/user# `lsblk`  
NAME        MAJ:MIN RM   SIZE RO TYPE MOUNTPOINTS
loop0         7:0    0   2.8G  1 loop /usr/lib/live/mount/rootfs/filesystem.squashfs
                                     /run/live/rootfs/filesystem.squashfs
sda           8:0    0 931.5G  0 disk  
├─sda1        8:1    0 931.5G  0 part  
└─sda2        8:2    0    32M  0 part  
nvme0n1     259:0    0 931.5G  0 disk  
├─nvme0n1p1 259:1    0   300M  0 part  
└─nvme0n1p2 259:2    0 931.2G  0 part  
root@debian:/home/user# `mkdir /mnt/chroot`
root@debian:/home/user# `mount -o subvol=@ /dev/nvme0n1p2 /mnt/chroot`
root@debian:/home/user# `mount /dev/nvme0n1p1 /mnt/chroot/boot/efi`
root@debian:/home/user# `for i in /dev /dev/pts /proc /sys /sys/firmware/efi/efivars /run; do mount -B $i /mnt/
chroot$i; done`
root@debian:/home/user# `chroot /mnt/chroot /bin/sh -c "mount -a || true && /bin/bash"`
root@debian:/# `grub-install`
Installing for x86_64-efi platform.
Installation finished. No error reported.
root@debian:/# `update-grub`
Generating grub configuration file ...
Found background image: splash.png
Found linux image: /boot/vmlinuz-6.5.0-1-amd64
Found initrd image: /boot/initrd.img-6.5.0-1-amd64
Found linux image: /boot/vmlinuz-6.4.0-4-amd64
Found initrd image: /boot/initrd.img-6.4.0-4-amd64
Found linux image: /boot/vmlinuz-6.4.0-3-amd64
Found initrd image: /boot/initrd.img-6.4.0-3-amd64
Warning: os-prober will be executed to detect other bootable partitions.
Its output will be used to detect bootable binaries on them and create new boot entries.
grub-probe: error: cannot find a GRUB drive for /dev/sda2.  Check your device.map.
Adding boot menu entry for UEFI Firmware Settings ...
Detecting snapshots ...
Found snapshot: 2023-09-16 09:39:47 | @snapshots/305/snapshot | post | apt |
Found snapshot: 2023-09-16 09:36:37 | @snapshots/304/snapshot | pre  | apt |
--SNIP-- |
Found 50 snapshot(s)
Unmount /tmp/grub-btrfs.jJM6HuJe4X .. Success
done
root@debian:/#


I rebooted and the system is running again.


Thanks a lot.
